cmd/cli/kubectl-kyverno/utils/common/common.go

```diff
@@ -701,55 +701,82 @@ func GetPoliciesFromPaths(fs billy.Filesystem, dirPath []string, gitBranch strin
 
 // GetResourceAccordingToResourcePath - get resources according to the resource path
 func GetResourceAccordingToResourcePath(fs billy.Filesystem, resourcePaths []string,
-	cluster bool, policies []kyvernov1.PolicyInterface, dClient dclient.Interface, namespace string, policyReport bool, isGit bool, policyResourcePath string,
+	cluster bool, policies []kyvernov1.PolicyInterface, dClient dclient.Interface, namespace string, policyReport bool, resourceGitBranch string, policyResourcePath string,
 ) (resources []*unstructured.Unstructured, err error) {
-	if isGit {
-		resources, err = GetResourcesWithTest(fs, policies, resourcePaths, isGit, policyResourcePath)
-		if err != nil {
-			return nil, sanitizederror.NewWithError("failed to extract the resources", err)
+	if len(resourcePaths) > 0 && resourcePaths[0] == "-" {
+		if IsInputFromPipe() {
```

Comment:
Cool. Was pipe input provided as a requirement explicitly? Or you picked it from elsewhere in nctl?